src/full/Agda/Interaction/BasicOps.hs

```diff
@@ -1132,8 +1135,20 @@ introTactic pmLambda ii = do
       r <- splitLast CoInductive tel pat
       case r of
         Left err -> return []
-        Right cov ->
-           mapM showUnambiguousConName $ concatMap (conName . scPats) $ splitClauses cov
+        Right cov -> mapM conWithArgs $ catMaybes $ map (conName . scPats) $ splitClauses cov
+      where
+        conWithArgs ch = do
+             cn <- showUnambiguousConName ch
+             sia <- showImplicitArguments
+             return $ unwords $
+                         cn : [ case v of
+                                    Hidden ->      "{?}"
+                                    Instance _ -> "{{?}}"
+                                    NotHidden ->    "?"
+                              | v <- map getHiding $ conFields ch
+                              , v == NotHidden || sia ]
```

Comment:
Suggestion: write this comprehension as a `mapMaybe`:
```
(cn :) $ flip mapMaybe (map getHiding $ conFields ch) $ \case
  NotHidden  -> Just "?"
  Hidden     -> if sia then Just "{?}" else Nothing
  Instance _ -> Nothing
```
Structuring code without implicit invariants (e.g. that `v` cannot be `Hidden` unless `sia`) makes it easier to verify it by just local reasoning.
(This thinking comes out of experience with verification.  E.g. think about how you would write correct-by-construction code in Agda, and then transfer the principles to writing Haskell.)